docs/en/tutorials/getting-started.rst

```diff
@@ -228,33 +227,196 @@ entity definition:
          * @var string
          */
         protected $name;
+    }
 
-        public function getId()
+When creating entity classes, all of the fields should be protected or private (not public).
+
+Adding the Entity behaviors
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You have two options to define to define methods in your entity:
+**getters/setters**, or **mutators and DTOs**,
+respectively for **anemic models** or **rich models**.
+
+**Anemic models: Getters and setters**
+
+The most popular method is to create two kinds of methods to
+**read** (getter) and **update** (setter) the object's properties.
+
+Some fields such as ``$id`` are unlikely to be changed, so it's ok to
+omit them.
+
+.. note::
+
+    Setters are unrelated to Doctrine itself, because the ORM does not
+    use them to change the values of the fields. It instead relies on
+    the Reflection API and does not run either the constructor or other
+    methods.
+
+This method is mostly used when you want to focus on behavior-less
+entities, and when you need to move all your business logic in your
+services rather than in the objects themselves.
+
+It is a common convention which makes it possible to expose each field
+of your entity to external services, while allowing you to keep type
+safety in place.
+
+Such approach is a good choice for RAD (rapid application development),
+but may bring problems later down the road, because providing such an
+easy way to modify any field in your entity means the entity itself cannot
+ensure it's in valid state. Having entity in invalid state is dangerous,
+because it's one step away from being implicitly saved in database, thereby
+leading to corrupted or inconsistent data in your storage.
+
+.. note::
+
+    This method, although very common, is inappropriate for Domain Driven
+    Design (DDD), because in such design, methods should be named according
+    to actual business operations, and entities should be valid anytime.
```

Comment:
> and entities should be valid anytime.

And business invariants should be maintained both in the application state (entities in this case) and in the database, with no space for data corruption.